Show summary of Examine search results

bjarnef opened the pull request:
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description
In a current working project, we have a custom index and advanced search/lucene queries often returning many results.
We could inspect the request, but I think it would be nice to see to number of results returned and it can be used to compare with the expected results based on the dataset.

https://github.com/umbraco/Umbraco-CMS/assets/2919859/559555fa-6923-416e-864c-211b33fb63cb

nul800sebastiaan commented:
Thanks @bjarnef! Looks like a small but useful addition indeed! We'll have a look soon. 👍

bjarnef commented:
Actually I think we may want a slightly different format.

So instead of "Showing {number-of-page-items} of {total-items}` then "Showing {first-page-item}-{number-of-page-items} of {total-items}`, e.g. "Showing 51-100 of 597 result(s)".

abjerner commented:
Would it make sense to show the page numbers as well? In Skybrud Redirects, I'm using the following format:

> Showing 1-11 of 37 – Page 1 of 4

bjarnef commented:
I have updated the paging message.

![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/c7cc5aa1-23e5-499d-ad0b-be4f5b6a8ad2)

I also removed the hardcoded page size value of 20 in the controller and returned the actual page size from the API endpoint instead in case we change that sometime later.

nul800sebastiaan commented:
Thanks very much for the discussion and the updates. Looks great and tests great, thanks for the work @bjarnef! 💪
